src/EnergyPlus/PipeHeatTransfer.hh

```diff
@@ -311,116 +305,75 @@ namespace PipeHeatTransfer {
 			EnvironmentHeatLossRate( 0.0 ),
 			EnvHeatLossEnergy( 0.0 ),
 			VolumeFlowRate( 0.0 )
-		{}
-
-	};
 
-	// Object Data
-	extern Array1D< PipeHTData > PipeHT;
-	extern Array1D< PipeHeatTransferReport > PipeHTReport;
-
-	// Functions
-
-	void
-	SimPipesHeatTransfer(
-		int const EquipType,
-		std::string const & EquipName, // name of the Pipe Heat Transfer.
-		int & EqNum, // index in local derived types for external calling
-		bool const InitLoopEquip,
-		bool const FirstHVACIteration // component number
-	);
+		{}
 
-	//==============================================================================
+		static
+		std::shared_ptr< PlantComponent > factory( int objectType, std::string objectName );
```

Comment:
I'm being pedantic again, but i like the name create. Also not sure about that shared_ptr. Finally, maybe you could pass in the idf text or IdfObject or something. That way your factory doesn't have to access some global data to do the job.